rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/server/serialization/CatalogRequestBuilder.java

```diff
@@ -0,0 +1,128 @@
+        } else {
+            builder.append("/resources?");
+            appendParameters(builder,
+                             paramPair(NAME_PARAM, this.catalogObjectAction.getNodeSourceName()),
+                             paramPair(KIND_PARAM, this.catalogObjectAction.getKind()),
+                             paramPair(COMMIT_MESSAGE_PARAM, urlEncodedCommitMessage),
+                             paramPair(OBJECT_CONTENT_TYPE_PARAM, this.catalogObjectAction.getObjectContentType()));
+        }
+        return builder.toString();
+    }
+
+    private String getUrlEncodedCommitMessageOrFail() {
+        try {
+            return URLEncoder.encode(this.catalogObjectAction.getCommitMessage(), "UTF-8");
+        } catch (UnsupportedEncodingException e) {
+            throw new IllegalArgumentException("Commit message cannot be encoded properly", e);
+        }
+    }
+
+    private void appendParameters(StringBuilder builder, String... paramPairs) {
+        builder.append(Arrays.stream(paramPairs).collect(Collectors.joining("&")));
+    }
+
+    public CloseableHttpResponse postNodeSourceRequestToCatalog(String sessionId, String fullUri,
+            HttpServletResponse response)
+            throws NoSuchAlgorithmException, KeyStoreException, KeyManagementException, IOException {
+        HttpPost postNodeSource = buildCatalogRequest(sessionId, fullUri);
+        return getHttpClientBuilder().build().execute(postNodeSource);
+    }
+
+    private String paramPair(String param, String value) {
+        return param + "=" + value;
+    }
+
+    private HttpPost buildCatalogRequest(String sessionId, String fullUri) {
+        String boundary = "---------------" + UUID.randomUUID().toString();
+        HttpPost post = new HttpPost(fullUri);
+        post.addHeader("Accept", "application/json");
+        post.addHeader("Content-Type",
+                       org.apache.http.entity.ContentType.MULTIPART_FORM_DATA.getMimeType() + ";boundary=" + boundary);
+        post.addHeader("sessionId", sessionId);
+
+        MultipartEntityBuilder builder = MultipartEntityBuilder.create();
+        builder.setBoundary(boundary);
+        builder.setMode(HttpMultipartMode.BROWSER_COMPATIBLE);
+        builder.addPart("file", new FileBody(this.catalogObjectAction.getNodeSourceJsonFile()));
+        post.setEntity(builder.build());
+        return post;
+    }
+
+    private HttpClientBuilder getHttpClientBuilder()
+            throws KeyStoreException, NoSuchAlgorithmException, KeyManagementException {
+        SSLContextBuilder sslContextBuilder = new SSLContextBuilder();
+        sslContextBuilder.loadTrustMaterial(null, new TrustSelfSignedStrategy());
+        SSLConnectionSocketFactory sslSocketFactory = new SSLConnectionSocketFactory(sslContextBuilder.build(),
+                                                                                     SSLConnectionSocketFactory.ALLOW_ALL_HOSTNAME_VERIFIER);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this use of "ALLOW_ALL_HOSTNAME_VERIFIER"; it is deprecated. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)